internal/allocator/allocator.go

```diff
@@ -310,17 +310,9 @@ func (a *Allocator) sortedPoolsForService(svc *v1.Service) []*config.Pool {
 	if svc == nil {
 		return priorityPools
 	}
-	for _, nsPoolName := range a.pools.ByNamespace[svc.Namespace] {
-		if nsPool, ok := a.pools.ByName[nsPoolName]; ok {
-			if !nsPool.AutoAssign {
-				continue
-			}
-			if nsPool.ServiceAllocations.Priority > 0 {
-				priorityPools = append(priorityPools, nsPool)
-				continue
-			}
-			noPriorityPools = append(noPriorityPools, nsPool)
-		}
+	if len(a.pools.ByNamespace[svc.Namespace]) > 0 {
+		priorityPools, noPriorityPools = a.sortedNamespacePoolsForService(svc)
```

Comment:
thinking again about this, now the priority / no priority is useless as long as we handle the compare function properly, because at the end we exit from here with one single list

Reply:
Also, instead of reimplementing it, I'd leverage the isPoolCompatibleWithService in both this loop and the one below. Basically the idea is "we see that this matches, let's see if it's compatible.

I'd do the same in the loop below even if it's logically reduntant now and burns a few cycles, because in the future we might extend the "isPoolCompatible" function with more checks.

Reply:
yes, correct, we don't need `noPriorityPools` and it can be done with sort function itself.
for other comment, though these methods has some duplicate piece of code and each one is used for different purpose. I think existing structure looks to be cleaner.

Reply:
I think having two loops (one for namespaces and the other for services) that check the match on each side, and then the compatibility is more symmetric and easier to reason about. 

Every loop should be like "check if this pool matches upon criteria x, and check if matches completely". This is easier to follow because we don't need to think that we already checked the namespaces in the previous loop, and  will make things easier if we add a third criteria.


So, what I am proposing is to use the `isPoolCompatibleWithService` function while we check in `getNamespacePoolsForService` and also in the `range a.pools.ByServiceSelector` loop. Also, since the former will be slimmer because of that, we can consider either removing the wrapper function (or wrapping the service selector logic in another function too, it feels asymmetric).

Reply:
OK, I see your point. changed it accordingly, Hope it looks better now.

Reply:
yes! Much better